### PR TITLE
Recaptcha

### PR DESCRIPTION
**Pull request by HighPingg**
This PR adds Recaptcha v3 to the share story and subscribe forms.

### CODE REVIEW

File: app/share/views.py [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 
+from app.constants.subscribe_status import VALID, EMAIL_INVALID, EMAIL_TAKEN, PHONE_TAKEN
 from app.constants import RECAPTCHA_STRING
-from app.lib.utils import create_story, create_user, create_subscriber
+from app.lib.utils import create_story, create_user, create_subscriber, verify_subscriber
 from app.models import Tags
 from app.share import share
 from app.share.forms import StoryForm
 
+from config import Config
```
Comment (johnyu95) [inline app/share/views.py]: Can you remove this and use the `current_app['RECAPTCHA_PRIVATE_KEY']` style instead

File: app/share/views.py [hunk 2]
```diff
@@ -21,16 +25,47 @@ def new():
             email = form.user_email.data
             phone = form.user_phone.data
 
+            # Verify recaptcha token and return error if failed
+            recaptcha_response = requests.post(url=f'https://www.google.com/recaptcha/api/siteverify?secret={ Config.RECAPTCHA_PRIVATE_KEY }&response={ request.form["g-recaptcha-response"] }').json()
+            if recaptcha_response['success'] is False or recaptcha_response['score'] < 0.5:
+                flash(Markup('Recaptcha failed to validate you!'), category='danger')
```
Comment (johnyu95) [inline app/share/views.py]: Change flash message to be "Recaptcha failed, please try again."

File: app/subscribe/views.py [hunk 3]
```diff
@@ -15,15 +21,44 @@ def subscribe():
...
+
+            elif recaptcha_response['success'] is False or recaptcha_response['score'] < 0.5:
+                flash(Markup('Recaptcha failed to validate you!'), category='danger')
+
+            else:
+                # Check entered email and phone number
+                verify_email = verify_subscriber(email, phone)
+
+                if verify_email == EMAIL_TAKEN:
+                    flash(Markup('You are already subscribed with that email!'), category='warning')
```
Comment (johnyu95) [inline app/subscribe/views.py]: Change to "The email you entered is already subscribed, please try another email."

File: app/subscribe/views.py [hunk 4]
```diff
@@ -15,15 +21,44 @@ def subscribe():
...
+                    return render_template('subscribe/subscribe.html', form=form,
+                                           RECAPTCHA_PUBLIC_KEY=Config.RECAPTCHA_PUBLIC_KEY)
+
+                elif verify_email == PHONE_TAKEN:
+                    flash(Markup('You are already subscribed with that phone number!.'), category='warning')
```
Comment (johnyu95) [inline app/subscribe/views.py]: Change to "The phone number you entered is already subscribed, please use another phone number".

**Review by johnyu95 — CHANGES_REQUESTED** [review]

File: app/share/views.py (hunk 1 above)
Comment (HighPingg) [reply app/share/views.py]: done

File: app/share/views.py (hunk 2 above)
Comment (HighPingg) [reply app/share/views.py]: done

File: app/subscribe/views.py (hunk 3 above)
Comment (HighPingg) [reply app/subscribe/views.py]: done

File: app/subscribe/views.py (hunk 4 above)
Comment (HighPingg) [reply app/subscribe/views.py]: done

**Review by johnyu95 — APPROVED** [review]
Looks good, good job!

**Pull request merged**